chia/wallet/puzzles/p2_singleton_via_delegated_puzzle.clsp

```diff
@@ -32,10 +40,15 @@
   (defun for_parent_amounts (my_puzhash list_of_parent_amounts total)
     (if list_of_parent_amounts
         (cons_announcements_to_output
+          ; [REVIEW - optimization] why do we care if they have the same puzzle hash as us.
+          ; [^] _They_ should only care that _we_ are the same puzhash as them (so they don't get spent by some rando).
+          ; [^] This as it sits blocks potentially folding things that could be meltable in.
+          ; [^] The alternative would be to pass in a list of coin_id/amount pairs and have the asserting p2_singletons assert that their amount is accurate.
           (calculate_coin_id (f (f list_of_parent_amounts)) my_puzhash (f (r (f list_of_parent_amounts))))  ; asserts that all the coins have the same puzzle as me
           (for_parent_amounts my_puzhash (r list_of_parent_amounts) (+ total (f (r (f list_of_parent_amounts)))))
         )
         (list
+          ; [REVIEW - comments] I think higher would be fine, there's probably no burning need for aggregation on the part of an honest party
```

Comment:
The worry is about dust storms from dishonest parties, but we also want to prevent DoS via aggregation too.

Reply:
Right that's what I'm saying, dust is usually coming from a dishonest party so honest parties only need to do a couple of aggregations at most.  Trying to pick up the dust would be annoying, but how likely are you really to need to do that?